### DISCUSSION TITLE
Correct way of implementing a Status Bar

### DISCUSSION THREAD

**Discussion by alexandru-cazacu** [ocornut/imgui#3966] [Old/Archived discussions]
Hi everyone .

I'm thinkering with ImGui to see how much it can be customizable.
At the moment I'm trying to implement a status bar without too much success.

By status bar i mean something like a menu bar but at the bottom of the host window.

I tried the following approaches:

First:

- Duplicate BeginMainMenuBar()
- Position it at the bottom of the host window
- This leaves a gap under the main menu bar and cause some flickering.
- The status bar is renderer on top of existing windows.

Second:

- Manually create a dockspace with a vertical size of -30px (for the status bar).
- This causes some rendering artifacts where the status bar would be.

Can someone give me a hint on what is the best approach to take?

Apologies for my English.

**Comment by ocornut** [comment]
Hello,

Do you need a single status bar for the whole viewport, or a status bar in a floating window?

The rest of your message is too ambiguous for me to provide an answer. 

It's currently not easy to create non-scrolling area in _floating windows_ but we'll be working on a way on do it. Currently for _floating windows_ a workaround may be to push a clip rect and create a child window positioned at an absolute position.

**Reply by alexandru-cazacu** [reply]
Hi,

I need a single status bar for the whole viewport. Something like the main menu bar, but at the bottom of the viewport.

It should take it's own space instead of covering existing windows.

Is the clip rect approach good for this case or is there a better solution?

**Reply by ocornut** [reply]
> I need a single status bar for the whole viewport.

Then that should be possible by using code similar to `BeginMainMenuBar()`, it is only a matter of positioning/sizing your window. 

> Is the clip rect approach good for this case or is there a better solution?

That's unrelated, the cliprect is for adding status bar for floating windows which have their own scrolling region. If you create a window solely dedicated to carry the status bar it's easier.

You'll need to clarify those statements:

"This leaves a gap under the main menu bar and cause some flickering."
"The status bar is renderer on top of existing windows."

In the meanwhile I think I will try to provide a helper for this and have `BeginMainMenuBar()` use it so it will be easier to achieve what you want.

**Reply by ocornut** [reply]
Worked out some refactoring a58271c079e1e47b6913d8f611d6f219f0a1399f

**Reply by ocornut** [reply]
I suggest moving this conversation to #3518 (hadn't realized you were the same person who posted there, giving much more context too)

**Reply by alexandru-cazacu** [reply]
Thanks! It worked perfecly.

Ok. I'm closing this one in favor of #3518.
